test/test_inputs.py

```diff
@@ -48,31 +46,28 @@ def test_input_object_not_contained_in_document(self):
         assert input in doc.objects
 
     def test_unbounded_inputs(self):
-        doc = sbol3.Document()
+        protocol, doc = labop.Protocol.initialize_protocol()
```

Comment:
I don't love this pattern, because it doesn't feel very Pythonic. The Pythonic thing to do is to use constructors. This is starting to look like Java.  

However, I sympathize with what you are trying to accomplish.  The current approach for initializing a protocol is Byzantine: first construct a Document, then construct a Protocol, then add the Protocol to the Document...

FWIW I have been scheming other ways to hide all of the Document housekeeping.

Reply:
Coming soon in another PR